app/src/androidTest/java/org/kiwix/kiwixmobile/main/KiwixMainRobot.kt

```diff
@@ -0,0 +1,115 @@
+import org.kiwix.kiwixmobile.Findable.ViewId
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.help.HelpRobot
+import org.kiwix.kiwixmobile.help.help
+import org.kiwix.kiwixmobile.localFileTransfer.LocalFileTransferRobot
+import org.kiwix.kiwixmobile.localFileTransfer.localFileTransfer
+import org.kiwix.kiwixmobile.nav.destination.library.LibraryRobot
+import org.kiwix.kiwixmobile.nav.destination.library.OnlineLibraryRobot
+import org.kiwix.kiwixmobile.nav.destination.library.library
+import org.kiwix.kiwixmobile.nav.destination.library.onlineLibrary
+import org.kiwix.kiwixmobile.nav.destination.reader.ReaderRobot
+import org.kiwix.kiwixmobile.nav.destination.reader.reader
+import org.kiwix.kiwixmobile.page.bookmarks.BookmarksRobot
+import org.kiwix.kiwixmobile.page.bookmarks.bookmarks
+import org.kiwix.kiwixmobile.page.history.HistoryRobot
+import org.kiwix.kiwixmobile.page.history.history
+import org.kiwix.kiwixmobile.settings.SettingsRobot
+import org.kiwix.kiwixmobile.settings.settingsRobo
+import org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString
+import org.kiwix.kiwixmobile.webserver.ZimHostRobot
+import org.kiwix.kiwixmobile.webserver.zimHost
+
+fun kiwixMain(func: KiwixMainRobot.() -> Unit) =
+  KiwixMainRobot().applyWithViewHierarchyPrinting(func)
+
+class KiwixMainRobot : BaseRobot() {
+
+  init {
+    isVisible(ViewId(R.id.navigation_container))
+  }
+
+  fun clickReaderOnBottomNav(func: ReaderRobot.() -> Unit) {
+    clickOn(ViewId(R.id.readerFragment))
+    reader(func)
+  }
+
+  fun clickLibraryOnBottomNav(func: LibraryRobot.() -> Unit) {
+    clickOn(ViewId(R.id.libraryFragment))
+    library(func)
+    pressBack()
+  }
+
+  fun clickFileTransferIcon(func: LocalFileTransferRobot.() -> Unit) {
+    clickOn(ViewId(R.id.get_zim_nearby_device))
+    localFileTransfer(func)
+  }
+
+  fun clickDownloadOnBottomNav(func: OnlineLibraryRobot.() -> Unit) {
+    clickOn(ViewId(R.id.downloadsFragment))
+    onlineLibrary(func)
+    pressBack()
+  }
+
+  fun clickBookmarksOnNavDrawer(func: BookmarksRobot.() -> Unit) {
+    clickOn(Text(getResourceString(R.string.bookmarks)))
+    bookmarks(func)
+    pressBack()
+    pressBack()
+  }
+
+  fun clickHistoryOnSideNav(func: HistoryRobot.() -> Unit) {
+    clickOn(Text(getResourceString(R.string.history)))
+    history(func)
+    pressBack()
+    pressBack()
+  }
+
+  fun clickHostBooksOnSideNav(func: ZimHostRobot.() -> Unit) {
+    clickOn(Text(getResourceString(R.string.menu_host_books)))
+    zimHost(func)
+    pressBack()
+  }
+
+  fun clickSettingsOnSideNav(func: SettingsRobot.() -> Unit) {
+    clickOn(Text(getResourceString(R.string.menu_settings)))
+    settingsRobo(func)
+    pressBack()
+  }
+
+  fun clickHelpOnSideNav(func: HelpRobot.() -> Unit) {
+    clickOn(Text(getResourceString(R.string.menu_help)))
+    help(func)
+    pressBack()
+  }
+
+  fun clickSupportKiwixOnSideNav() {
+    clickOn(Text(getResourceString(R.string.menu_support_kiwix)))
```

Comment:
I replaced all these `Text(getResourceString(...)` with TextId and it worked fine so lets do that